File: escape_codes.js
// Colors
const colors = {
  backBlack: '\x1b[40m',
  backRed: '\x1b[41m',
  backGreen: '\x1b[42m',
  backYellow: '\x1b[43m',
  backBlue: '\x1b[44m',
  backPurple: '\x1b[45m',
  backCyan: '\x1b[46m',
  backWhite: '\x1b[107m',
  backOrange: '\x1b[48;5;208m',
  backDarkRed: '\x1b[101m',
  backDarkGreen: '\x1b[102m',
  backDarkYellow: '\x1b[103m',
  backDarkBlue: '\x1b[104m',
  backDarkPurple: '\x1b[105m',
  backDarkCyan: '\x1b[106m',
  backDarkGray: '\x1b[100m',
  backDarkOrange: '\x1b[48;5;202m',
  backLightRed: '\x1b[101m',
  backLightGreen: '\x1b[102m',
  backLightYellow: '\x1b[103m',
  backLightBlue: '\x1b[104m',
  backLightPurple: '\x1b[105m',
  backLightCyan: '\x1b[106m',

  foreBlack: '\x1b[30m',
  foreRed: '\x1b[31m',
  foreGreen: '\x1b[32m',
  foreYellow: '\x1b[33m',
  foreBlue: '\x1b[34m',
  forePurple: '\x1b[35m',
  foreCyan: '\x1b[36m',
  foreGray: '\x1b[37m',
  foreWhite: '\x1b[97m',
  foreOrange: '\x1b[38;5;208m',
  foreDarkRed: '\x1b[91m',
  foreDarkGreen: '\x1b[92m',
  foreDarkYellow: '\x1b[93m',
  foreDarkBlue: '\x1b[94m',
  foreDarkPurple: '\x1b[95m',
  foreDarkCyan: '\x1b[96m',
  foreDarkGray: '\x1b[90m',
  foreDarkOrange: '\x1b[38;5;202m',
  foreLightRed: '\x1b[91m',
  foreLightGreen: '\x1b[92m',
  foreLightYellow: '\x1b[93m',
  foreLightBlue: '\x1b[94m',
  foreLightPurple: '\x1b[95m',
  foreLightCyan: '\x1b[96m',
};

// Styles
const styles = {
  reset: '\x1b[0m',
  bold: '\x1b[1m',
  dim: '\x1b[2m',
  underline: '\x1b[4m',
  blink: '\x1b[5m',
  reverse: '\x1b[7m',
  hidden: '\x1b[8m',
  strikethrough: '\x1b[9m',
  magical: '\x1b[6m',
  slowBlink: '\x1b[25m',
  fastBlink: '\x1b[6m',
  italic: '\x1b[3m',
  overline: '\x1b[53m',
  doubleUnderline: '\x1b[21m',
  framed: '\x1b[51m',
  encircled: '\x1b[52m',
  overlined: '\x1b[55m',
  boldFaint: '\x1b[20m',
  positiveImg: '\x1b[27m',
};

// Cursor
const cursor = {
  up: '\x1b[A',
  down: '\x1b[B',
  forward: '\x1b[C',
  backward: '\x1b[D',
  pos: '\x1b[H',
  clear: '\x1b[2J',
  clearForward: '\x1b[0J',
  clearBackward: '\x1b[1J',
  clearLine: '\x1b[2K',
  clearLineForward: '\x1b[0K',
  clearLineBackward: '\x1b[1K',
  savePos: '\x1b[s',
  restorePos: '\x1b[u',
};

module.exports = { colors, styles, cursor };

File: escape_codes.java
public class AnsiEscapeCodes {
    // Colors
    public static final String BACK_BLACK = "\u001B[40m";
    public static final String BACK_RED = "\u001B[41m";
    public static final String BACK_GREEN = "\u001B[42m";
    public static final String BACK_YELLOW = "\u001B[43m";
    public static final String BACK_BLUE = "\u001B[44m";
    public static final String BACK_PURPLE = "\u001B[45m";
    public static final String BACK_CYAN = "\u001B[46m";
    public static final String BACK_WHITE = "\u001B[107m";
    public static final String BACK_ORANGE = "\u001B[48;5;208m";

    public static final String FORE_BLACK = "\u001B[30m";
    public static final String FORE_RED = "\u001B[31m";
    public static final String FORE_GREEN = "\u001B[32m";
    public static final String FORE_YELLOW = "\u001B[33m";
    public static final String FORE_BLUE = "\u001B[34m";
    public static final String FORE_PURPLE = "\u001B[35m";
    public static final String FORE_CYAN = "\u001B[36m";
    public static final String FORE_WHITE = "\u001B[97m";
    public static final String FORE_ORANGE = "\u001B[38;5;208m";

    // Styles
    public static final String RESET = "\u001B[0m";
    public static final String BOLD = "\u001B[1m";
    public static final String DIM = "\u001B[2m";
    public static final String UNDERLINE = "\u001B[4m";
    public static final String BLINK = "\u001B[5m";
    public static final String REVERSE = "\u001B[7m";
    public static final String HIDDEN = "\u001B[8m";
    public static final String STRIKETHROUGH = "\u001B[9m";
    public static final String MAGICAL = "\u001B[6m";
    public static final String SLOW_BLINK = "\u001B[25m";
    public static final String FAST_BLINK = "\u001B[6m";
    public static final String ITALIC = "\u001B[3m";
    public static final String OVERLINE = "\u001B[53m";
    public static final String DOUBLE_UNDERLINE = "\u001B[21m";
    public static final String FRAMED = "\u001B[51m";
    public static final String ENCIRCLED = "\u001B[52m";
    public static final String OVERLINED = "\u001B[55m";
    public static final String BOLD_FAINT = "\u001B[20m";
    public static final String POSITIVE_IMG = "\u001B[27m";

    // Cursor
    public static final String UP = "\u001B[A";
    public static final String DOWN = "\u001B[B";
    public static final String FORWARD = "\u001B[C";
    public static final String BACKWARD = "\u001B[D";
    public static final String POS = "\u001B[H";
    public static final String CLEAR = "\u001B[2J";
    public static final String CLEAR_FORWARD = "\u001B[0J";
    public static final String CLEAR_BACKWARD = "\u001B[1J";
    public static final String CLEAR_LINE = "\u001B[2K";
    public static final String CLEAR_LINE_FORWARD = "\u001B[0K";
    public static final String CLEAR_LINE_BACKWARD = "\u001B[1K";
    public static final String SAVE_POS = "\u001B[s";
    public static final String RESTORE_POS = "\u001B[u";
    public static final String NEXT_LINE = "\u001B[E";
    public static final String PREV_LINE = "\u001B[F";
}
